src/plugins/dashboard/public/plugin.tsx

```diff
@@ -308,6 +329,27 @@ export class DashboardPlugin
       stopUrlTracker();
     };
 
+    const registerDashboardListSource = (
+      pluginName: string,
+      listProviderFn: () => Observable<DashboardListItems>
+    ) => {
+      this.dashboardListSources.push({ name: pluginName, listProviderFn });
+    };
+
+    const registerDashboardItemCreator = (creator: DashboardCreator) => {
+      this.dashboardItemCreators.push(creator);
```

Comment:
nit: Can you validate that a dashboard item creator of the same name does not already exist?

Reply:
Will do.